src/rc-socket.js

```diff
@@ -30,293 +21,341 @@ var root = this;
  * ---------------------------------------------------------------------------*/
 
 /**
- * This behaves like a WebSocket in every way, except if it fails to connect,
- * or it gets disconnected, it will use an exponential backoff until it
- * succesfully connects again.
+ * @global
+ * @public
+ * @constructor
+ *
+ * @name RcSocket
+ * @desc This behaves like a WebSocket in every way, except if it fails to
+ * connect, or it gets disconnected, it will use an exponential backoff until
+ * it succesfully connects again.
  *
  * It is API compatible with the standard WebSocket API.
  *
  * @example
  * var ws = new ReconnectingWebsocket(wss://host);
  *
- * @public
- * @constructor
  * @param {String} url - Url to connect to.
  * @param {String|Array} protocols - Optional subprotocols.
  */
 var RcSocket = function (url, protocols) {
-  // Defaults
-  this.debug    = false;
-  this.timeout  = 2500;
-  this.maxRetry = 1000;
-  this.delay    = 100;
-  this.logger   = console.debug;
+  this.url = url;
+  this.protocols = protocols;
 
-  // State
-  this.unload   = false;
-  this.forced   = false;
-  this.timedOut = false;
+  this.debug = false;
+  this.timeout = 2500;
+  this.maxRetry = 1000;
+  this.delay = 100;
+  this.logger = console;
+
+  this.hasUnloaded = false;
+  this.hasOpened = false;
+  this.wasForced = false;
+  this.isRetrying = false;
+  this.isRefreshing = false;
```

Comment:
I think this implementation is fine.  

But I'd like to hear your thoughts on using a [FSM](http://stackoverflow.com/a/1371654) to represent the states that the socket is in.  I realize this is a larger discussion, but just putting some thoughts down.

```
// socket state
var ss = {
 UNLOADED : 1, 
 OPENED : 2, 
 RETRYING : 3, 
 REFRESHING : 4
};

var SocketSM = {
  ss.UNLOADED : {
    'onclose' :                     // evt string
      this.closeHandler       // function which should be called on this event
  }
...
};
```